src/App.tsx

```diff
@@ -1,61 +1,130 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [newTodos, setNewTodos] = useState(todos);
+  const [errorEmptyTitle, setErrorEmptyTitle] = useState(false);
+  const [errorEmptyselectedUser, seterrorEmptyselectedUser] = useState(false);
+
+  // eslint-disable-next-line @typescript-eslint/no-shadow
+  const correctionTodo = (title: string, selectedUser: number) => {
+    let correctTitle = '';
+
+    if (title) {
+      // eslint-disable-next-line no-plusplus
+      for (let i = 0; i < title.length; i++) {
+        if ((title[i].toLocaleLowerCase() !== title[i].toLocaleUpperCase())
+          || Number.isInteger(+title[i])) {
+          correctTitle += title[i];
+        }
+      }
+    }
+
+    const newTodo = {
+      title: correctTitle,
+      userId: selectedUser,
+      id: Math.max(...newTodos.map(todo => todo.id)) + 1,
+      completed: false,
+      user: getUser(selectedUser),
+    };
+
+    setNewTodos(arr => [...arr, newTodo]);
+  };
```

Comment:
it looks like `handleTodoAdd`. `correctionTodo` doesn't explain real functionality of this code.